apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -152,3 +171,68 @@ export const Contacts = {
 		return contactId;
 	},
 };
+
+export async function createContact(params: CreateContactParams): Promise<string> {
+	const { name, emails, phones, customFields = {}, contactManager, channels, unknown } = params;
+
+	if (contactManager) {
+		const contactManagerUser = await Users.findOneById<Pick<IUser, 'roles'>>(contactManager, { projection: { roles: 1 } });
+		if (!contactManagerUser) {
+			throw new Error('error-contact-manager-not-found');
+		}
+		if (!contactManagerUser.roles || !Array.isArray(contactManagerUser.roles) || !contactManagerUser.roles.includes('livechat-agent')) {
+			throw new Error('error-invalid-contact-manager');
+		}
```

Comment:
You can use `users.findOneAgentById` and remove the manual `livechat-agent` check here